src/components/Icon/index.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import ClassNames from 'classnames';
+import styles from '@/components/Icon/Icon.css';
+
+import IconSearch from '@/assets/images/icons/web/search-simple.svg';
+
+type Category = 'search';
+
+interface Props {
+  description?: string;
+  heigth?: string;
+  name?: Category;
+  title?: string;
+  variablesClassName?: string;
+  width?: string;
+}
+
+const icons = {
+  search: IconSearch
+};
+
+const Icon: React.FC<Props> = props => {
+  const { description, title, variablesClassName, ...iconProps } = props;
+  const IconType = icons[props.name || ''];
+  const descClass = 'offscreen-desc';
+  const buttonClass = 'offscreen-button';
+
+  return (
+    <button className={ClassNames(variablesClassName, styles[buttonClass])}>
```

Comment:
Since the `description` is optional, I'd suggest adding an `aria-hidden={Boolean(description) ? "false" : "true"}` attr to the button in case the description is missing, this is due to if there's no description at all, screen readers will not be able to identify what this icon is for, with the aria-hidden we hide the content to screen readers in case the icon is meant to be only for visual purposes